### ISSUE TITLE
Switch slim.learning.train to use tf.train.MonitoredTrainingSession, and expose sessionrunhooks

### ISSUE THREAD

**Issue by krishnashankar** [tensorflow/tensorflow#18600]
Hi! 

I've been getting warnings when using slim.learning.train about switching to tf.train.MonitoredTrainingSession. Is there a plan to make the switch?

Thanks!

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by krishnashankar** [comment]
Have I written custom code No
OS Platform and Distribution Ubuntu 16.04
TensorFlow installed from virtualenv
TensorFlow version 1.7.0
Bazel version 0.10
CUDA/cuDNN version 9.0/7.0
GPU model and memory Titan X
Exact command to reproduce N/A

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 22 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 37 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 67 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 82 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 97 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 112 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 127 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 142 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by wt-huang** [comment]
@krishnashankar Could you try TensorFlow 1.11.0 and see if this issue is still there?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by wt-huang** [comment]
Closing due to inactivity, feel free to reopen if problem persists.
